DOC-6252 - provide annotated source code for use in documentation

mikereiche opened the pull request:
Motivation
==========
To ensure that example code in documentation is accurate.

Changes
=======
When example code from devguides-examples is to be used in documentation
it requires (a) to be in a specific directory structure for Antora; and
(b) to be annotated for Antora. As well, we wish to maintain the code
as it was under devguide-examples, without any Antora annotations.
To satisfy all these requirements, files required by the documentation
will be copied to docs-sdk-nodejs/modules/devguide/examples/nodejs/...
and have the appropriate annotations added:
 // #tag::insert[]
 .
 .
 .
 // #end::insert[]
The asciidoctor (adoc) files can then reference the annotated portions
with includes like :
  include::devguide:example$nodejs/kv-operations.js[tag=insert,indent=0]
Note that the Antora identifier does not include a 'component' (repo),
as the kv-operations.js file is in *this* component.  The 'devguide'
refers to the name of the module where the file resides under.
Although only the files that are used in the docs need to be copied/moved
to docs-sdk-nodejs/modules/devguide/examples/nodejs/... , they
could/should all be moved for consistency.

Commit:
DOC-6252 - provide annotated source code for use in documentation

Motivation
==========
To ensure that example code in documentation is accurate.

Changes
=======
When example code from couchbaselabs/sdk-examples is to be used in documentation
it requires (a) to be in a specific directory structure for Antora; and
(b) to be annotated for Antora. As well, we wish to maintain the code
as it was under couchbaselabs/sdk-examples, without any Antora annotations.
To satisfy all these requirements, files required by the documentation
will be copied to docs-sdk-nodejs/modules/devguide/examples/nodejs/...
and have the appropriate annotations added:
 // #tag::insert[]
 .
 .
 .
 // #end::insert[]
The asciidoctor (adoc) files can then reference the annotated portions
with includes like :
  include::devguide:example$nodejs/kv-operations.js[tag=insert,indent=0]
Note that the Antora identifier does not include a 'component' (repo),
as the kv-operations.js file is in *this* component.  The 'devguide'
refers to the name of the module where the file resides under.
Although only the files that are used in the docs need to be copied/moved
to docs-sdk-nodejs/modules/devguide/examples/nodejs/... , they
could/should all be moved for consistency.

Commit:
DOC-6252 - provide annotated source code for use in documentation

Motivation
==========
To ensure that example code in documentation is accurate.

Changes
=======
When example code from devguides-examples is to be used in documentation
it requires (a) to be in a specific directory structure for Antora; and
(b) to be annotated for Antora. As well, we wish to maintain the code
as it was under devguide-examples, without any Antora annotations.
To satisfy all these requirements, files required by the documentation
will be copied to docs-sdk-nodejs/modules/devguide/examples/nodejs/...
and have the appropriate annotations added:
 // #tag::insert[]
 .
 .
 .
 // #end::insert[]
The asciidoctor (adoc) files can then reference the annotated portions
with includes like :
  include::devguide:example$nodejs/kv-operations.js[tag=insert,indent=0]
Note that the Antora identifier does not include a 'component' (repo),
as the kv-operations.js file is in *this* component.  The 'devguide'
refers to the name of the module where the file resides under.
Although only the files that are used in the docs need to be copied/moved
to docs-sdk-nodejs/modules/devguide/examples/nodejs/... , they
could/should all be moved for consistency.

RichardSmedley commented on modules/devguide/examples/striptags.sh:
```diff
@@ -21,8 +21,9 @@ egrep -v '^striptags.sh|^nodejs/node_modules|^nodejs/package-lock.json' |\
   else
```
Are you going to update this for the new sdk-examples site?

Commit:
DOC-6252 - provide annotated source code for use in documentation

Motivation
==========
To ensure that example code in documentation is accurate.

Changes
=======
When example code from couchbaselabs/sdk-examples is to be used in documentation
it requires (a) to be in a specific directory structure for Antora; and
(b) to be annotated for Antora. As well, we wish to maintain the code
as it was under couchbaselabs/sdk-examples, without any Antora annotations.
To satisfy all these requirements, files required by the documentation
will be copied to docs-sdk-nodejs/modules/devguide/examples/nodejs/...
and have the appropriate annotations added:
 // #tag::insert[]
 .
 .
 .
 // #end::insert[]
The asciidoctor (adoc) files can then reference the annotated portions
with includes like :
  include::devguide:example$nodejs/kv-operations.js[tag=insert,indent=0]
Note that the Antora identifier does not include a 'component' (repo),
as the kv-operations.js file is in *this* component.  The 'devguide'
refers to the name of the module where the file resides under.
Although only the files that are used in the docs need to be copied/moved
to docs-sdk-nodejs/modules/devguide/examples/nodejs/... , they
could/should all be moved for consistency.

RichardSmedley reviewed (approved):
LGTM
